### DISCUSSION TITLE
docs: createParamDecorator question

### DISCUSSION THREAD

**Discussion by carmas123** [mikro-orm/mikro-orm#3335] [Q&A]
Hi, I migrate from TypeORM to MikroORM and now I have a problem using it expecially into a Param Decorator:
To handle some request parameters that need to access to database with TypeORM I can access to entity data using this approach:
`MyEntity.findOne()`

This canno be done with MikroORM and using createParamDecorator I cannot found any method to inject or use MikroORM entity manager.

Please can anyone help me to solve my problem?

**Comment by B4nan** [comment]
You will need to share more details. What cannot be done?

**Reply by carmas123** [reply]
```
import {
  BadRequestException,
  createParamDecorator,
  ExecutionContext,
} from '@nestjs/common';
import { User } from '../../users/users.entity';
import { TokenDataIntf } from '../interfaces';

export const UserParam = createParamDecorator(
  async (required = true, ctx: ExecutionContext): Promise<User> => {
    const req = ctx.switchToHttp().getRequest();
    const user = req.user as TokenDataIntf;
    let result: User = undefined;
    if (user && user.id) {
      // Now we find the user to add information to parameter
      result = await User.findOne(user.id);
    }
    if (required && !result) {
      throw new BadRequestException('Invalid user');
    }
    return result;
  },
);
```

This is my code with TypeORM, as you can see with TypeORM I can do that: User.findOne(user.id);
Now there is a way to do that also with MikroORM?

**Reply by B4nan** [reply]
`User.findOne` just uses global instance (a huge antipattern if you ask me), you can do that here as well - either store it on some global place, wherever you want, or if you are using the `RequestContext` helper, you can get the contextual fork from that via `RequestContext.getEntityManager()`.

**Reply by carmas123** [reply]
Hi @B4nan sometime `RequestContext.getEntityManager()` return `undefined` and this is a big problem because I use it as ParamDecoration (with NestJS) and the same ParamDecoration works on one endpoint and not on others.

Please there is another way to use the `RequestContext.getEntityManager()` to work fine without return undefined?

**Reply by B4nan** [reply]
It returns `undefined` if you call it outside of the async context created by `RequestContext.create()`, this isn't "fixable" from out end, this is about the flow of your application. Might be caused by a bug in nest, I was seeing weird things with module inheritance and the order of execution of middlewares, but wasn't able to reproduce properly. I also heard from people having broken async contexts (which results in this too) via some nest 3rd party modules (usually those that register some middleware).

You could try registering the middleware explicitly like the realworld example app does:

https://github.com/mikro-orm/nestjs-realworld-example-app/blob/master/src/app.module.ts#L32-L39

And in such case you will need to disable the automatic middleware:

https://github.com/mikro-orm/nestjs-realworld-example-app/blob/master/src/mikro-orm.config.ts.example#L19
